package.json

```diff
@@ -19,6 +19,7 @@
     "test:unit-ci": "karma start karma.ci.js --single-run",
     "test:image-diff": "MDC_COMMIT_HASH=$(git rev-parse --short HEAD) MDC_BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD) mocha --require ts-node/register --require babel-core/register --ui tdd --timeout 60000 test/screenshot/diff-suite.tsx",
     "test:screenshots": "docker run -it --rm --cap-add=SYS_ADMIN -e MDC_GCLOUD_SERVICE_ACCOUNT_KEY=\"${MDC_GCLOUD_SERVICE_ACCOUNT_KEY}\" mdcreact/screenshots /bin/sh -c 'git checkout .; git checkout master; git pull; npm i; /home/pptruser/material-components-web-react/test/screenshot/start.sh; sleep 40s; npm run test:image-diff'",
+    "test:types": "npm run build && node scripts/release/cp-pkgs.js && rm -rf packages/*/node_modules && npm install packages/ripple && ls -d ./packages/*/ | xargs -n1 -I{} npm install {} && node test/types/gen-index.js && tsc --project test/types || echo 'TODO(issues/857): Fix existing errors before requiring this to pass'",
```

Comment:
- this test looks specifically to ripple...this should be more generic.
- what is npm install {}?

Reply:
It was clearer before it was condensed onto a single line (it had comments).

`ripple` has to be installed first because otherwise it gets installed from online before the local install, and then weird conflicts happen.

It's not `npm install {}`, it's `ls -d ./packages/*/ | xargs -n1 -I{} npm install {}` which installs all packages in the repo directory :-)

Reply:
`ls -d ./packages/*/` = list all subdirectories under packages
`| xargs -n1 -I{}` = pipe into `xargs`, run once per input line, substitute `{}` for the input line
`npm install {}` = install package substituting `{}` for the directory path